### PR TITLE
CNDE-1819: Addition of Update and Key Generation to Lab Test Result and Morbidity Report Post-Processing, Wiring of Hepatitis Lab Columns 

### PR DESCRIPTION
**Pull request by upasanapattnaik-eq**
## Notes
Featured Changes:

- Update operation is added for target dimensions in sp_d_labtest_result_postprocessing and sp_d_morbidity_report_postprocessing. Sprocs support both insert and update. 
- Tested on latest Observation Service. 
- Key generation for TEST_RESULT_GROUPING, RESULT_COMMENT_GROUP and  MORB_RPT_USER_COMMENT.
- Removal of global temporary tables and handling dynamic column assignment in Morb Report sproc using dynamic SQL. 

Additional:

- Hepatitis Datamart Additional Column Wiring for 
1. INIT_NND_NOT_DT
3. NOT_SUBMIT_DT
5. EVENT_DATE
7. EVENT_DATE_TYPE

_sp_d_labtest_result_postprocessing_
- LAB_TEST_RESULT
- TEST_RESULT_GROUPING
-  RESULT_COMMENT_GROUP
-  LAB_RESULT_VAL
-  LAB_RESULT_COMMENT

_sp_d_morbidity_report_postprocessing:_

- MORBIDITY_REPORT
- MORBIDITY_REPORT_EVENT
- MORB_RPT_USER_COMMENT

## JIRA

- **CNDE-1819**: [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNDE-1819?atlOrigin=eyJpIjoiNmI5NWFkYjg5MzFmNGU0Yzg4MjYxMmE4Yjk5ODBiYTciLCJwIjoiaiJ9)
- - **CNDE-1691**: [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNDE-1691?atlOrigin=eyJpIjoiZjk3Mzc0ZGNiYTk3NGNmNTgzOGQ5YjMwMTI0OTNkMjgiLCJwIjoiaiJ9)

## Checklist

- [x] PR focuses on a single story.
- [ ] New unit tests added and ensured they pass.
- [x] Service has been tested in local and it works as expected.
- [ ] Documentation has been updated for this code change (if needed).
- [ ] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [ ] New feature
- [ ] Breaking change

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [ ] Does the `gradle build` pass in your local? 
- [ ] Is the `gradle build` logs attached?

### CODE REVIEW

**Review by sveselev — APPROVED** [review]
LGTM!

File: liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql [hunk 1]
```diff
@@ -0,0 +1,2609 @@
+CREATE OR ALTER PROCEDURE dbo.sp_hepatitis_datamart_postprocessing
+    @phc_id nvarchar(max),
+    @pat_ids nvarchar(max) = '',
```
Comment (rseshadri99) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql:3]: @upasanapattnaik-eq why @pat_ids are defaulted to empty string? Do we have a logic to look for empty string before it is used in downstream sql?

File: liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql [hunk 2]
```diff
@@ -0,0 +1,2609 @@
...
+             ON PAGE_CASE.PHYSICIAN_KEY = P.PROVIDER_KEY
+                 LEFT OUTER JOIN
+             dbo.D_PROVIDER AS INVGTR WITH(NOLOCK)
+             ON PAGE_CASE.INVESTIGATOR_KEY = INVGTR.PROVIDER_KEY
+                 LEFT OUTER JOIN
+             dbo.D_ORGANIZATION AS REPTORG WITH(NOLOCK)
+             ON PAGE_CASE.ORG_AS_REPORTER_KEY = REPTORG.ORGANIZATION_KEY
+        ORDER BY HEP_PAT_PROV_INV_KEY;
+
+        UPDATE #TMP_HEP_PAT_PROV
```
Comment (rseshadri99) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql:1236]: @upasanapattnaik-eq  could this be moved to the previous Select

File: liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql [hunk 3]
```diff
@@ -0,0 +1,2609 @@
...
+        SET PHYS_NAME = RTRIM(LTRIM(PHYSICIAN_first_nm));
+
+        UPDATE #TMP_HEP_PAT_PROV
```
Comment (rseshadri99) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql:1239]: @upasanapattnaik-eq  could this be moved to the previous Select

File: liquibase-service/src/main/resources/db/rdb_modern/routines/017-sp_d_labtest_result_postprocessing-002.sql [hunk 4]
```diff
@@ -0,0 +1,1450 @@
...
+                                    FROM #TMP_New_Lab_Result_Comment ST1
+                                    WHERE ST1.lab_test_uid = LRV.lab_test_uid
+                                    ORDER BY ST1.lab_test_uid,ST1.lab_result_txt_seq
+                                    FOR XML PATH ('')
+                                ), 2, 2000) v_lab_result_val_txt
+        INTO #TMP_New_Lab_Result_Comment_grouped
+        FROM #TMP_New_Lab_Result_Comment LRV;
+
+
+        UPDATE #TMP_New_Lab_Result_Comment
```
Comment (rseshadri99) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/017-sp_d_labtest_result_postprocessing-002.sql:304]: @upasanapattnaik-eq Move the temp table update statement logic to the Select SQL. Across the file I see a lot of Updates that are smaller transformations

**Review by rseshadri99 — CHANGES_REQUESTED** [review]
@upasanapattnaik-eq Move the temp table update statement logic to the Select SQL. Across these files I see a lot of Updates that are smaller transformations

File: liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql (hunk 1 above)
Comment (upasanapattnaik-eq) [reply liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql:3]: It is not used downstream as of right now.  The service sends both phc_id and associated pat ids. 
As an improvement ticket,  both will be incorporated to enforce updates/inserts to Hepatitis datamart.

File: liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql (hunk 3 above)
Comment (upasanapattnaik-eq) [reply liquibase-service/src/main/resources/db/rdb_modern/routines/013-sp_hepatitis_datamart_postprocessing-003.sql:1239]: Will take the individual updates as enhancement items.

**Pull request merged**